### PR TITLE
add list pod

### PR DESCRIPTION
**Pull request by xlgao-zju**
A part of #8 

Signed-off-by: Xianglin Gao <xlgao@zju.edu.cn>

---
This change is [Reviewable](https://reviewable.kubernetes.io/reviews/kubernetes/frakti/20)

### CODE REVIEW

File: pkg/hyper/hyper.go [hunk 1]
```diff
@@ -155,7 +155,58 @@ func (h *Runtime) PodSandboxStatus(podSandboxID string) (*kubeapi.PodSandboxStat
...
+		items = append(items, &kubeapi.PodSandbox{
+			Id:        &pod.PodID,
+			Metadata:  podSandboxMetadata,
+			Labels:    pod.Labels,
+			State:     &state,
+			CreatedAt: &pod.CreatedAt,
+		})
+	}
+
+	return items, nil
```
Comment (feiskyer) [inline pkg/hyper/hyper.go]: items should be sorted by creation time (newest first) before return.

File: pkg/hyper/hyper.go [hunk 2]
```diff
@@ -155,7 +155,58 @@ func (h *Runtime) PodSandboxStatus(podSandboxID string) (*kubeapi.PodSandboxStat
...
+				continue
+			}
+
+			if filter.State != nil && state != filter.GetState() {
+				continue
+			}
+		}
+
+		if filter != nil && filter.LabelSelector != nil &&
+			!inMap(filter.LabelSelector, pod.Labels) {
```
Comment (feiskyer) [inline pkg/hyper/hyper.go]: Labels filtering should be moved up into `if filter != nil {  }`

File: pkg/hyper/hyper.go [hunk 3]
```diff
@@ -155,7 +155,58 @@ func (h *Runtime) PodSandboxStatus(podSandboxID string) (*kubeapi.PodSandboxStat
...
+		glog.Errorf("GetPodList failed: %v", err)
+		return nil, err
+	}
+
+	items := make([]*kubeapi.PodSandbox, 0, len(pods))
+	for _, pod := range pods {
+		state := toPodSandboxState(pod.Status)
+
+		if filter != nil {
+			if filter.Name != nil && pod.PodName != filter.GetName() {
```
Comment (feiskyer) [inline pkg/hyper/hyper.go]: pod.PodName should be split first to get the kubernetes podName.

**Comment by xlgao-zju** [comment]
@feiskyer updated

File: pkg/hyper/helper.go [hunk 4]
```diff
@@ -39,6 +40,8 @@ const (
 	fraktiAnnotationLabel = "io.kubernetes.frakti.annotations"
 )
 
+type Items []*kubeapi.PodSandbox
```
Comment (feiskyer) [inline pkg/hyper/helper.go]: nit: `Items` should be renamed to something more specific, e.g. `sandboxByCreated`

**Comment by feiskyer** [comment]
@xlgao-zju Generally lgtm, with only one nit.

**Comment by xlgao-zju** [comment]
@feiskyer fixed.

**Comment by feiskyer** [comment]
Thanks. LGTM.

**Pull request merged**